### DISCUSSION TITLE
iFAN script modification fail

### DISCUSSION THREAD

**Discussion by MagOO33** [RavenSystem/esp-homekit-devices#1955] [MEPLHAA Scripts]
Hello,
I've tried to modify the iFAN03/04 script with RF to have a 3 steps fan instead of the false continuous one in home app.
I've tried this :
{"c":{"z":12,"r":[{"n":0,"s":9600}],"ul":[8,8],"l":13,"b":[{"g":0,"t":5}]},"a":[{"t":65,"e":3,"0":{"r":[{"g":15},{"g":12},{"g":14}]},"y0":[{"v":1,"0":{"r":[{"g":15,"v":1,"i":0.5},{"g":12},{"g":14,"i":0.6}]}},{"v":2,"0":{"r":[{"g":15,"v":1,"i":0.5},{"g":12,"i":0.6},{"g":14,"i":0.6}]}},{"v":3,"0":{"r":[{"g":15,"v":1},{"g":12},{"g":14,"v":1}]}}],"es":[{"s":5,"1":{"r":[{"g":9}]},"0":{"r":[{"g":9,"v":1}]},"b":[{"g":0}]}]},{"t":80,"h":0,"n":24,"pt":[["AA",5]],"dt":[2,0],"y0":[{"v":260,"r":1,"0":{"m":[[7002,1]]}},{"v":517,"r":1,"0":{"m":[[7002,1]]}},{"v":6,"r":1,"0":{"m":[[7001,1],[-2]]}},{"v":263,"r":1,"0":{"m":[[7001,1],[-2,1]]}},{"v":520,"r":1,"0":{"m":[[7001,1],[-2,2]]}},{"v":777,"r":1,"0":{"m":[[7001,1],[-2,3]]}},{"v":1034,"r":1,"0":{"m":[[7001,1],[-1,4]]}},{"v":265,"r":1,"0":{"m":[[7004,1]]}},{"v":266,"r":1,"0":{"m":[[7003,1]]}}]},{"h":0,"i":0.1,"1":{"u":[{"v":"AA550104000005"}]}},{"h":0,"i":0.1,"1":{"u":[{"v":"AA550101000002"}]}},{"h":0,"i":0.1,"1":{"u":[{"v":"AA550107000008"}]}},{"h":0,"i":0.1,"1":{"u":[{"v":"AA550106000007"}]}}]}

It works in home app with 3 steps and light, but the RF is only controlling the light.
This script works well for RF 👍 but it has no steps in home app :
 {"c":{"z":12,"r":[{"n":0,"s":9600}],"ul":[8,8],"l":13,"b":[{"g":0,"t":5}]},"a":[{"t":65,"0":{"r":[{"g":15},{"g":12},{"g":14}]},"y0":[{"v":1,"0":{"r":[{"g":15,"v":1,"i":0.5},{"g":12},{"g":14,"i":0.6}]}},{"v":33,"0":{"r":[{"g":15,"v":1,"i":0.5},{"g":12,"i":0.6},{"g":14,"i":0.6}]}},{"v":67,"0":{"r":[{"g":15,"v":1},{"g":12},{"g":14,"v":1}]}}],"es":[{"s":5,"1":{"r":[{"g":9}]},"0":{"r":[{"g":9,"v":1}]},"b":[{"g":0}]}]},{"t":80,"h":0,"n":24,"pt":[["AA",5]],"dt":[2,0],"y0":[{"v":260,"r":1,"0":{"m":[[7002,1]]}},{"v":517,"r":1,"0":{"m":[[7002,1]]}},{"v":6,"r":1,"0":{"m":[[7001,1],[-2]]}},{"v":263,"r":1,"0":{"m":[[7001,1],[-2,30],[-2,101]]}},{"v":520,"r":1,"0":{"m":[[7001,1],[-2,60],[-2,101]]}},{"v":777,"r":1,"0":{"m":[[7001,1],[-2,100],[-2,101]]}},{"v":1034,"r":1,"0":{"m":[[7001,1],[-1,4]]}},{"v":265,"r":1,"0":{"m":[[7004,1]]}},{"v":266,"r":1,"0":{"m":[[7003,1]]}}]},{"h":0,"i":0.1,"1":{"u":[{"v":"AA550104000005"}]}},{"h":0,"i":0.1,"1":{"u":[{"v":"AA550101000002"}]}},{"h":0,"i":0.1,"1":{"u":[{"v":"AA550107000008"}]}},{"h":0,"i":0.1,"1":{"u":[{"v":"AA550106000007"}]}}]}

I don't understand the RF part of the script enough to find any clue.
Could anyone help me ?
Thanks

**Comment by RavenSystem** [comment]
These are the target speeds from RF remote (30%, 60% and 100%):
[-2,30]
[-2,60]
[-2,100]

Replace them with 1, 2 and 3:
[-2,1]
[-2,2]
[-2,3]

**Reply by MagOO33** [reply]
Thank you,
I think I've done this.
I replaced this :
{"v":263,"r":1,"0":{"m":[[7001,1],[-2,30],[-2,101]]}},
{"v":520,"r":1,"0":{"m":[[7001,1],[-2,60],[-2,101]]}},
{"v":777,"r":1,"0":{"m":[[7001,1],[-2,100],[-2,101]]}},
by this :
{"v":263,"r":1,"0":{"m":[[7001,1],[-2,1]]}},
{"v":520,"r":1,"0":{"m":[[7001,1],[-2,2]]}},
{"v":777,"r":1,"0":{"m":[[7001,1],[-2,3]]}},
But I only deleted the three instances of ,[-2,101]
Is it the problem ?
